design-toolkit/components/src/components/kanban/kanban.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2025 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+'use client';
+import 'client-only';
+
+import {
+  type CollisionDetection,
+  closestCenter,
+  DndContext,
+  type DragEndEvent,
+  DragOverlay,
+  type DragStartEvent,
+  PointerSensor,
+  pointerWithin,
+  rectIntersection,
+  useSensor,
+  useSensors,
+} from '@dnd-kit/core';
+import { createContext, useContext, useState } from 'react';
+import { parseDropTarget, useKanban } from '@/components/kanban/context';
+import { CardInnerStyles, KanbanStyles } from './styles';
+import type { KanbanProps } from './types';
+
+const { container, cardHeader, cardTitle, cardBody } = KanbanStyles();
+
+// Context for sharing active drag state
+export const DragContext = createContext<{ activeId: string | null }>({
+  activeId: null,
+});
+
+export const useDragContext = () => useContext(DragContext);
```

Comment:
need error handling: match how useKanban throws error if not used within the KanbanProvider


```suggestion
export const useDragContext = () => {
  const context = useContext(DragContext);
  if (!context) {
    throw new Error('useDragContext must be used within Kanban component');
  }
  return context;
};
```